modules/schema/src/main/java/org/apache/ignite/internal/schema/BinaryRow.java

```diff
@@ -44,12 +47,11 @@ public interface BinaryRow {
     /** Get has value flag: {@code true} if row has non-null value, {@code false} otherwise. */
     boolean hasValue();
 
+    int length();
+
     /** Get ByteBuffer slice representing the binary tuple. */
     ByteBuffer tupleSlice();
 
-    /** Get byte array of the row. */
-    byte[] bytes();
-
     /** Returns the representation of this row as a Byte Buffer. */
     ByteBuffer byteBuffer();
```

Comment:
> It is also used to serialize rows as parts of Raft commands.

This part we can remove. It would be better to convert these objects into a BinaryRowMessage. Less memory manipulations, better performance. Ideally, there should zero memory copying between layers. It's not achievable, bet we should try to be as close to is as possible.

Reply:
Do you want me to do that in this PR? That's quite a big change

Reply:
No, we should do it in a separate issue

Reply:
I've created a ticket and added TODOs: https://issues.apache.org/jira/browse/IGNITE-19937